lib/FreeRTOS/freertos-plus-mqtt/freertos_mqtt_agent.c

```diff
@@ -1387,11 +1396,14 @@ MQTTStatus_t MQTTAgent_Init( MQTTContextHandle_t mqttContextHandle,
     static StaticQueue_t staticQueueStructure;
 
     /* The command queue should not have been created yet. */
-    configASSERT( commandQueue == NULL ); /*_RB_ Needs to change so two contexts can be used in the same agent. */
-    commandQueue = xQueueCreateStatic( MQTT_AGENT_COMMAND_QUEUE_LENGTH,
-                                       sizeof( Command_t * ),
-                                       staticQueueStorageArea,
-                                       &staticQueueStructure );
+    if( commandQueue == NULL )  /*_RB_ Needs to change so two contexts can be used in the same agent. */
+    {
+        LogDebug( ( "Creating command queue." ) );
+        commandQueue = xQueueCreateStatic( MQTT_AGENT_COMMAND_QUEUE_LENGTH,
+                                           sizeof( Command_t * ),
+                                           staticQueueStorageArea,
+                                           &staticQueueStructure );
```

Comment:
Should users have the ability to allocate this queue from the heap?

Reply:
That will be part of the queue abstraction, which we haven't designed yet. This change was necessary here to allow multiple connections though